### PR TITLE
Malayalam Translation

### PR DESCRIPTION
**Pull request by sagarvd01**
Added index.md and sidebar.md

### CODE REVIEW

File: _i18n/ml-in/changes.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: "QMKയിൽ വന്ന മാറ്റങ്ങൾ"
+layout: qmk-title
+permalink: /ml/changes/
+---
+
+{% for change in site.changes %}
```
Comment (noroadsleft) [inline _i18n/ml-in/changes.md]: This is a change that recently went into our main branch.

```suggestion
{% for change in site.changes reversed %}
```

File: _i18n/ml-in/changes.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+title: "QMKയിൽ വന്ന മാറ്റങ്ങൾ"
+layout: qmk-title
+permalink: /ml/changes/
+---
```
Comment (noroadsleft) [inline _i18n/ml-in/changes.md]:
```suggestion
lang: "ml"
---
```

File: _i18n/ml-in/index.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: "AVR, ARM അടിസ്ഥാനമാക്കിയുള്ള കീബോർഡുകൾക്ക് വേണ്ടിയുള്ള ഓപ്പൺ സോഴ്സ് ഫേംവെയർ"
+permalink: /ml-in/
+lang: ml-in
```
Comment (noroadsleft) [inline _i18n/ml-in/index.md]:
```suggestion
lang: "ml"
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
All of these files should go in `_i18n/ml/`, rather than `_i18n/ml-in/`.

**Comment by sagarvd01** [comment]
Hi @noroadsleft I made the changes and also updated folder from `ml-in` to `ml`

File: _i18n/ml/index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ഈ പ്രോജക്ടുകൾ  അവരുടെ ഔദ്യോഗിക ഫേംവെയർ ആയി QMK - ഉപയോഗിക്കുന്നു. നിങ്ങളുടെത് ലിസ്റ്റിൽ കാണുന്നില്ലെങ്കിൽ [ഞങ്ങളെ ബന്ധപ്പെടുക](https://github.com/qmk/qmk.fm/issues/new) 
+
+QMK മറ്റ് [നിരവധി കമ്മ്യൂണിറ്റികളുടെ കീബോർഡുകളുമായും പ്രോജക്ടുകളുമായും](/keyboards/) ഉപയോഗിക്കാൻ സാധിക്കും. 
+
+നിങ്ങളുടെ കീബോർഡിൽ QMK എങ്ങനെ ഉപയോഗിക്കാമെന്നതിനെ പറ്റി കൂടുതലറിയാൻ ഈ [ലിങ്ക്](/powered/) ഉപയോഗിക്കുക.
+
+## മെയിന്റൈൻ ചെയ്യുന്നവർ
+
+[Hasu](https://github.com/tmk)വിന്റെയും അതുപോലെ കമ്മ്യൂണിറ്റിയിലെ മറ്റുള്ളവരുടെയും സഹായത്തോടെ QMK നിർമിച്ചതും പരിപാലിക്കുന്നതും OLKBയിലെ Jack Humbert ആണ്. OLKB പ്രോഡക്ടിന്റെ ഫേംവെയർ മെയിന്റൈൻ ചെയ്യുന്നത് [Jack Humbert](https://github.com/jackhumbert), Ergodox EZ by [ZSA Technology Labs](https://github.com/zsa), പിന്നെ Clueboard by [Zach White](https://github.com/skullydazed) എന്നിവരാണ്.
+ .
```
Comment (noroadsleft) [inline _i18n/ml/index.md]:
```suggestion
```

File: _i18n/ml/index.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+---
+title: "AVR, ARM അടിസ്ഥാനമാക്കിയുള്ള കീബോർഡുകൾക്ക് വേണ്ടിയുള്ള ഓപ്പൺ സോഴ്സ് ഫേംവെയർ"
+permalink: /ml/
+lang: ml
+---
+Atmel AVR നും ARM കൺട്രോളറിനും അഥവാ [OLKB product line](http://olkb.com), [ErgoDox EZ](http://www.ergodox-ez.com) കീബോർഡ്, [Clueboard product line](http://clueboard.co/) തുടങ്ങിയവയ്ക്ക് ഉപകാരപ്രദമായ കുറേ സവിശേഷതകളുള്ള, 
+[tmk\_keyboard firmware](http://github.com/tmk/tmk_keyboard) നെ അടിസ്ഥാനമാക്കിയുള്ള കീബോർഡ് ഫേംവെയറാണ് ഇത്.
```
Comment (noroadsleft) [inline _i18n/ml/index.md:7]: Lines 6 and 7 seem to be in reverse order of how they are on the English site.

File: _i18n/ml/sidebar.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+[![Build Status](https://travis-ci.org/qmk/qmk_firmware.svg?branch=master)](https://travis-ci.org/qmk/qmk_firmware)
+[![Discord](https://img.shields.io/discord/440868230475677696.svg)](https://discord.gg/Uq7gcHh)
+[![Docs Status](https://img.shields.io/badge/docs-ready-orange.svg)](https://docs.qmk.fm)
+[![GitHub contributors](https://img.shields.io/github/contributors/qmk/qmk_firmware.svg)](https://github.com/qmk/qmk_firmware/pulse/monthly)
+[![GitHub forks](https://img.shields.io/github/forks/qmk/qmk_firmware.svg?style=social&label=Fork)](https://github.com/qmk/qmk_firmware/)
+
+### Pages
+
+* [ഓൺലൈൻ കോൺഫിഗറേറ്റർ](https://config.qmk.fm)
+* [QMKയിൽ അടുത്തിടെ വന്ന മാറ്റങ്ങൾ](/changes/)
```
Comment (noroadsleft) [inline _i18n/ml/sidebar.md]:
```suggestion
* [QMKയിൽ അടുത്തിടെ വന്ന മാറ്റങ്ങൾ](/ml/changes/)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: _i18n/ml/index.md (hunk 5 above)
Comment (sagarvd01) [reply _i18n/ml/index.md:7]: Hi @noroadsleft , the reverse order is to get the meaningful sentence without any redundant words. If we convert it as it is to Malayalam, there will be some unnecessary words which will confuse people.

Comment (noroadsleft) [reply _i18n/ml/index.md:7]: @sagarvd01 Thanks for the explanation. Obviously I don't know Malayalam, but it just looked weird to me.

**Comment by noroadsleft** [comment]
Last request, and I'll merge this:

We need an entry added to https://github.com/qmk/qmk.fm/blob/gh-pages/_data/strings.yml for Malayalam.

The English block is https://github.com/qmk/qmk.fm/blob/2fd93bef627acdf2d6ebe95a53aceb1229e18bd7/_data/strings.yml#L2-L8

We'll need an equivalent entry for Malayalam added to this file.

**Comment by sagarvd01** [comment]
Hi @noroadsleft , I missed that one earlier.

Updated strings.yml. I saw the statement `<<: *DEFAULT_EN` in other language blocks and I believe it's fallback statement to english.

**Review by noroadsleft — APPROVED** [review]
Excellent, thanks!

**Comment by noroadsleft** [comment]
> Updated strings.yml. I saw the statement `<<: *DEFAULT_EN` in other language blocks and I believe it's fallback statement to english.

@sagarvd01 You are correct, it sets up the site so if the translated page isn't found, the English data is used.

**Pull request merged**